### ISSUE TITLE
Resource type image tag update is not propagated correctly to existing resources

### ISSUE THREAD

**Issue by stepanstipl** [concourse/concourse#773]
Hi there!

# Bug Report

If tag is changed for resource_type, this is not propagated correctly to already existing resources and these will keep using old version for check step.

let's take for example:
```
resource_types:
- name: test-resource
  type: docker-image
  source:
    repository: xxx/yyy
    tag: 1.0
```

and you update the pipeline to:
```
resource_types:
- name: test-resource
  type: docker-image
  source:
    repository: xxx/yyy
    tag: 2.0
```

Existing resources with this resource_type will still keep using the 1.0 image for check step. Workaround that seems to fix this is to remove the resource from the pipeline definition and then add it again.

* Concourse version: 2.4.0
* Deployment type (BOSH/Docker/binary): Docker

**Comment by vito** [comment]
This is expected. The `source` is just configuration for where to discover new versions; already-discovered versions will remain along with their history (which builds produced/consumed them, etc.) You should disable the versions you no longer want after changing the configuration.

**Comment by stepanstipl** [comment]
@vito I think we're not talking about the same thing, what I'm saying is that the `check` step will keep using the **old image** to discover new resources.

Please notice I'm talking about **resource_types:** definition, not normal resource definition.


So these two scenarios will produce different results, which I believe should not be the case:

a) update custom resource_type with a new image tag (1 step)

vs

b) - delete custom resource_type (1st step)
     - recreate custom resource_type with same name and new tag (2nd step)

The latter will make check step on the resource reflect the change and properly use new image when checking for new resources.

**Comment by vito** [comment]
ah sorry yea I misread. it should pick up the new image in a minute or two.
if not this should most definitely be fixed in #629

**Comment by stepanstipl** [comment]
Ok cool, thanks, currently it does not seem to pick up the new image at all in our case (or at least not within an hour or so).
